public/bitemp_graph_helper.js

```diff
@@ -144,8 +142,136 @@ function edit(uri) {
   }
 }
 
-function deleteDoc(uri) {
-  if (!uri) { // Not given a valid document uri
+function getTemporalColl(uri) {
+  
+  var docColl = $.ajax({
+    url: '/manage/v2/databases/Documents/temporal/collections?format=json',
+    uri: uri,
+    success: function(data, textStatus) {
+     console.log('Success');
+    },
+    error: function(jqXHR, textStatus, errorThrown) {
+      console.log('Problem');
+    },
+    async: false,
+  });
+
+ return JSON.parse(docColl.responseText);
+}
+
+function getDocColls(uri) {
+  var docColl = $.ajax({
+    url: '/v1/documents?uri='+uri+'&category=collections&format=json',
+    success: function(data, textStatus) {},
+    error: function(jqXHR, textStatus, errorThrown) {
+      console.log('problem');
+    },
+    async: false,
+  });
+
+ return JSON.parse(docColl.responseText);
+}
+
+/*
+@params: 
+collArr is an array of strings of collection names
+tempCollArr is an array of objects with 'nameref' properties
+*/
+function findCommonColl(collArr, tempCollArr) {
+  var response;
+  while (!response) {
+    for (var i in collArr) {
+      for (var j in tempCollArr) {
+        if (collArr[i] === tempCollArr[j]['nameref']) {
+          console.log('Match: ' + collArr[i]);
+          response = collArr[i];
+        }
+      }
+    }
+  }
+  return response;
+}
+
+var deleteDoc = function (chart) {
+//var checkTimeRanges = function (chart) {
+  var uri = chart.getLogicalURI();
+  var ajax = true;
+  if (!uri) {
+    return;
+  }
+  var collArr = getDocColls(uri);
+  var tempCollections = getTemporalColl(uri);
+  var tempCollArr = tempCollections['temporal-collection-default-list']['list-items']['list-item'];
+  
+  var tempColl;
+  if (collArr && tempCollArr) {
+    collArr = collArr['collections'];
+    tempColl = findCommonColl(collArr, tempCollArr);
+  }
+  
+  $.ajax( //Gets a temporal collection
+  {
+    url: 'http://localhost:3000/v1/resources/temporal-range?rs:collection='+tempColl,
```

Comment:
You can leave out the 'http://localhost:3000'. By doing so, it's one less place to change if the app is run on a different port. Also, suppose you were accessing the application from a remote server -- localhost wouldn't get where you want to go.